tests/playbooks/tests_macvlan.yml

```diff
@@ -0,0 +1,219 @@
+- hosts: all
+  vars: 
+    interface: lsrmcvl31
+    profile: "{{ interface }}"
+    lsr_fail_debug:
+      - __network_connections_result
+  tasks:
+    - debug:
+        msg: "this is: playbooks/tests_macvlan.yml"
+      tags:
+        - always
+
+      
+    - block:
+        - include_tasks: tasks/run_test.yml
+          vars:
+            lsr_description: I can create a macvlan connection 
+            lsr_setup:
+              - tasks/delete_interface.yml
+              - tasks/assert_device_absent.yml
+            lsr_test:
+              - tasks/create_macvlan_interface.yml
+            lsr_assert:
+              - tasks/assert_device_present.yml
+            lsr_cleanup:
+              - tasks/cleanup_profile+device.yml
+      tags:
+        - tests::macvlan:create
+      
+    - block:
+        - include_tasks: tasks/run_test.yml
+          vars:
+            lsr_description: I can create a macvlan interface without autoconnect
+            lsr_setup:
+              - tasks/delete_interface.yml
+              - tasks/assert_device_absent.yml
+            lsr_test:
+              - tasks/create_macvlan_interface_no_autoconnect.yml
+            lsr_assert:
+              - tasks/assert_device_present.yml
+            lsr_cleanup:
+              - tasks/cleanup_profile+device.yml
+      tags:
+        - tests::macvlan:create_without_autoconnect
+
+    - block:
+        - include_tasks: tasks/run_test.yml
+          vars:
+            lsr_description: I can remove an existing connection without taking it down
+            lsr_setup:
+              - tasks/create_macvlan_interface.yml
+            lsr_test:
+              - tasks/remove_profile.yml
+            lsr_assert:
+              - tasks/assert_device_absent.yml
+            lsr_cleanup:
+              - tasks/cleanup_profile+device.yml
+      tags:
+        - tests::macvlan:remove_up
+      
+    - block:
+        - include_tasks: tasks/run_test.yml
+          vars:
+            lsr_description: I can take an absent macvlan down and remove it.
+            lsr_setup:
+              - tasks/create_macvlan_interface.yml
+            lsr_test:
+              - tasks/remove+down_profile.yml
+            lsr_assert:
+              - tasks/assert_device_absent.yml
+            lsr_cleanup:
+              - tasks/cleanup_profile+device.yml
+      tags:
+        - tests::macvlan:remove_down
+    
+    - block:
+        - include_tasks: tasks/run_test.yml
+          vars:
+            lsr_description: I will not get an error when I try to remove an absent macvlan 
+            lsr_setup:
+              - tasks/create_macvlan_interface.yml
+              - tasks/remove+down_profile.yml
+            lsr_test:
+              - tasks/remove+down_profile.yml
+              - tasks/get_NetworkManager_NVRA.yml
+            lsr_assert_when:
+              - what: tasks/assert_device_absent.yml
+                when: "{{
+                  NetworkManager_NVRA != 'NetworkManager-1.18.4-3.el7.x86_64'
```

Comment:
Could you use a version comparison test here instead?  https://docs.ansible.com/ansible/latest/user_guide/playbooks_tests.html#version-comparison

Reply:
It seems we don't need this, since macvlan won't have a profile, so we only need to assert if the device is absent in every case...

Reply:
Remove the conditional assert for this remove twice test